nano/core_test/ledger_walker.cpp

```diff
@@ -0,0 +1,220 @@
+	EXPECT_EQ (walked_blocks_count, 1);
+}
+
+namespace nano
+{
+TEST (ledger_walker, genesis_account_longer)
+{
+	nano::system system{};
+	nano::node_config node_config (nano::get_available_port (), system.logging);
+	node_config.enable_voting = true;
+	node_config.receive_minimum = 1;
+
+	const auto node = system.add_node (node_config);
+
+	nano::ledger_walker ledger_walker{ node->ledger };
+	EXPECT_TRUE (ledger_walker.walked_blocks.empty ());
+	EXPECT_EQ (1, ledger_walker.walked_blocks.bucket_count ());
+	EXPECT_TRUE (ledger_walker.blocks_to_walk.empty ());
+
+	const auto get_number_of_walked_blocks = [&ledger_walker] (const auto & start_block_hash) {
+		std::size_t walked_blocks_count = 0;
+		ledger_walker.walk_backward (start_block_hash,
+		[&] (const auto & block) {
+			++walked_blocks_count;
+		});
+
+		return walked_blocks_count;
+	};
+
+	const auto transaction = node->ledger.store.tx_begin_read ();
+	nano::account_info genesis_account_info{};
+	ASSERT_FALSE (node->ledger.store.account.get (transaction, nano::nano_dev_account, genesis_account_info));
+	EXPECT_EQ (get_number_of_walked_blocks (genesis_account_info.open_block), 1);
+	EXPECT_EQ (get_number_of_walked_blocks (genesis_account_info.head), 1);
+
+	system.wallet (0)->insert_adhoc (nano::dev_genesis_key.prv);
+	for (auto itr = 1; itr <= 5; ++itr)
+	{
+		const auto send = system.wallet (0)->send_action (nano::dev_genesis_key.pub, nano::dev_genesis_key.pub, 1);
+		ASSERT_TRUE (send);
+		EXPECT_EQ (get_number_of_walked_blocks (send->hash ()), 1 + itr * 2 - 1);
+		ASSERT_TIMELY (3s, 1 + itr * 2 == node->ledger.cache.cemented_count);
+		ASSERT_FALSE (node->ledger.store.account.get (transaction, nano::nano_dev_account, genesis_account_info));
+		// TODO: check issue with account head
+		// EXPECT_EQ(get_number_of_walked_blocks (genesis_account_info.head), 1 + itr * 2);
+	}
+
+	EXPECT_TRUE (ledger_walker.walked_blocks.empty ());
+	EXPECT_EQ (1, ledger_walker.walked_blocks.bucket_count ());
+	EXPECT_TRUE (ledger_walker.blocks_to_walk.empty ());
+}
+
+}
+
+TEST (ledger_walker, cross_account)
+{
+	nano::system system{};
+	nano::node_config node_config (nano::get_available_port (), system.logging);
+	node_config.enable_voting = true;
+	node_config.receive_minimum = 1;
+
+	const auto node = system.add_node (node_config);
+
+	system.wallet (0)->insert_adhoc (nano::dev_genesis_key.prv);
+	ASSERT_TRUE (system.wallet (0)->send_action (nano::dev_genesis_key.pub, nano::dev_genesis_key.pub, 1));
+	ASSERT_TIMELY (3s, 3 == node->ledger.cache.cemented_count);
+
+	nano::keypair key{};
+	system.wallet (0)->insert_adhoc (key.prv);
+	ASSERT_TRUE (system.wallet (0)->send_action (nano::dev_genesis_key.pub, key.pub, 1));
+	ASSERT_TIMELY (3s, 5 == node->ledger.cache.cemented_count);
+
+	const auto transaction = node->ledger.store.tx_begin_read ();
+	nano::account_info account_info{};
+	ASSERT_FALSE (node->ledger.store.account.get (transaction, key.pub, account_info));
+
+	//    TODO: check issue with account head
```

Comment:
I'd prefer not adding it as it won't run, rather we could add an issue on GitHub to track this.

Reply:
These are a couple of extra type of checks for unit tests that for some reason do not work as I expected them to. The ledger walker is most likely going to be changed again soon anyways (in different PR) so I don't think this would sit around too much commented out, but yeah maybe I should just take it out? @clemahieu what do you think?